cmd/manager/main.go

```diff
@@ -107,87 +119,162 @@ func main() {
 		LeaderElectionID:       "c5baf8af.sigs.k8s.io",
 	}
 	if configFile != "" {
```

Comment:
i think it it better to check that configFile is empty and fail, and then continue with the rest of the code

Reply:
Agreed. Actually I think we should deprecate all command-line flags in favor of the config file. @mresvanis WDYT?

Reply:
I would suggest that we deprecate the command-line flags altogether and just have the config file as the only way  to configure the operator. 

Having one way only would ease development and maintenance and at the same time it would follow [what](https://github.com/kubernetes/community/blob/master/archive/wg-component-standard/component-config/README.md) the Kubernetes components (e.g. `kubelet`, `scheduler`) are doing, which is replacing command-line flags with the config file.